torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -355,13 +360,14 @@ def foreach_reduce(
                 reduce_output += partial_reduce_output
             post_reduce_stream = all_reduce_stream
             all_reduce_stream.wait_stream(reduce_scatter_stream)
-            with torch.cuda.stream(all_reduce_stream):
+            with device_handle.stream(all_reduce_stream):
                 dist.all_reduce(
                     reduce_output,
                     group=all_reduce_group,
-                    op=ReduceOp.AVG if predivide_factor is None else ReduceOp.SUM,
+                    # hpu need to add support fot AVG, just change it to proceed, will see accuracy issue
```

Comment:
nit: maybe can you not include this comment unless you are confident that you are going to fix it and then remove this comment later?

otherwise, I would prefer you track the HPU to-do separately from the main code

Reply:
@awgu sorry that was temp fix for verifying it on HPU, forgot to remove it. fixed in the latest patch.